src/Structures/CollectionTree.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Statamic\Eloquent\Structures;
+
+use Statamic\Eloquent\Structures\TreeModel as Model;
+use Statamic\Structures\CollectionTree as FileEntry;
+
+class CollectionTree extends FileEntry
+{
+    protected $model;
+
+    public static function fromModel(Model $model)
+    {
+        return (new static)
+            ->tree($model->tree)
+            ->handle($model->handle)
+            ->locale($model->locale)
+            ->initialPath($model->settings['initial_path'] ?? null)
+            ->syncOriginal()
+            ->model($model);
+    }
+
+    public function toModel()
+    {
+        return self::makeModelFromContract($this);
+    }
+
+    public static function makeModelFromContract($source)
+    {
+        $class = app('statamic.eloquent.navigations.tree_model');
```

Comment:
Back with another small observation - I think we'd like to use `statamic.eloquent.collections.tree_model`, right?

Reply:
Good spot, thank you.